hrms/payroll/doctype/employee_benefit_ledger/employee_benefit_ledger.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2025, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _
+from frappe.model.document import Document
+from frappe.utils import flt
+
+
+class EmployeeBenefitLedger(Document):
+	pass
+
+
+def create_employee_benefit_ledger_entry(ref_doc, args=None, delete=False):
+	EmployeeBenefitLedger = frappe.qb.DocType("Employee Benefit Ledger")
+
+	if delete:
+		(
+			frappe.qb.from_(EmployeeBenefitLedger)
+			.delete()
+			.where(EmployeeBenefitLedger.salary_slip == ref_doc.name)
+		).run()
+		return
```

Comment:
Since deletion is only doing this part and returning, and the rest of the function body isn't needed, you should separate the create and delete functions

create(delete=True) is a weird way to call this.